lib/school_house_web/views/html_helpers.ex

```diff
@@ -14,7 +14,9 @@ defmodule SchoolHouseWeb.HtmlHelpers do
   end
 
   def current_page_locale_path(%{request_path: request_path}, locale) do
-    String.replace(request_path, "/#{current_locale()}/", "/#{locale}/", global: false)
+    request_path
+    |> String.replace(~r/^\/#{current_locale()}(\/|$)/, "/#{locale}/", global: false)
+    |> String.trim_trailing("/")
```

Comment:
@jaimeiniesta  thanks for walking through that use case, I had not accounted for a link like `/essential` and did break it! 

The new changes will just check the beginning of the request path for the locale and replace it there if it exists. I two new tests for this as well (including your suggestion, thanks!). This handles the case where the user switches locales from the home page (`/es`) or from a blog post (`/blog/essential-elixir`).

A couple examples of switch from `es` to `ar` for example:
```
/es -> /ar
/blog/server-side-svg-charts-with-contex-and-liveview -> /blog/server-side-svg-charts-with-contex-and-liveview
/es/intermediate/escripts -> /ar/intermediate/escripts
```

Reply:
That's much better! I like that regexp use.